### PR TITLE
Added: UserController is done.

### PR DESCRIPTION
**Pull request by Khurshid0109**

### CODE REVIEW

File: src/MDC.Server.Service/DTOs/Users/UserForResultDto.cs [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 {
-    internal class UserForResultDto
-    {
-    }
+    public long Id { get; set; }
+    public string FirstName { get; set; }
+    public string LastName { get; set; }
+    public string Email { get; set; }
+    public string PhoneNumber { get; set; }
+    public string Password { get; set; }
```
[the remark this answers is not in the document]
Comment (Khurshid0109) [reply src/MDC.Server.Service/DTOs/Users/UserForResultDto.cs:10]: sorry
Must be in a hurry

File: src/MDC.Server.Service/DTOs/Users/UserForUpdateDto.cs [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+
+    [Required,MinLength(3)]
+    public string LastName { get; set; }
+
+    [MDCEmailAttribute]
+    public string Email { get; set; }
+    [PhoneAttribute]
+    public string PhoneNumber { get; set; }
+    [Required]
+    public string Password { get; set; }
```
Comment (Nozimjon04) [inline src/MDC.Server.Service/DTOs/Users/UserForUpdateDto.cs:18]: The password should not be updated here.  Another method should be written to update the password

File: src/MDC.Server.Service/Services/Users/UserService.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        var users = await _repository.SelectAll()
+             .ToListAsync();
+
+        return _mapper.Map<IEnumerable<UserForResultDto>>(users);
+    }
+
+    public async Task<UserForResultDto> RetrieveByEmailAsync(string email)
+    {
+        var user = await _repository.SelectAll()
+           .Where(u => u.Email==email)
```
Comment (Nozimjon04) [inline src/MDC.Server.Service/Services/Users/UserService.cs:84]: Email should be converted to lower to make case insensative

File: src/MDC.Server.Service/DTOs/Users/UserForUpdateDto.cs (hunk 2 above)
Comment (Khurshid0109) [reply src/MDC.Server.Service/DTOs/Users/UserForUpdateDto.cs]: Done

File: src/MDC.Server.Service/Services/Users/UserService.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+using MDC.Server.Service.Exceptions;
+using MDC.Server.Service.Interfaces.Users;
+using Microsoft.EntityFrameworkCore;
+using System.Runtime.Serialization;
+
+namespace MDC.Server.Service.Services.Users;
+public class UserService : IUserService
+{
+    private readonly IUserRepository _repository;
+    private readonly IMapper _mapper;
```
Comment (Nozimjon04) [inline src/MDC.Server.Service/Services/Users/UserService.cs:14]: Name _userRepository not repository

**Pull request merged**